frontend/src/components/common/Resource/ScaleButton.tsx

```diff
@@ -0,0 +1,222 @@
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { KubeObject } from '../../../lib/k8s/cluster';
+import { CallbackActionOptions, clusterAction } from '../../../redux/actions/actions';
+import { LightTooltip } from '../Tooltip';
+
+interface ScaleButtonProps {
+  item: KubeObject;
+  options?: CallbackActionOptions;
+}
+
+export default function ScaleButton(props: ScaleButtonProps) {
+  const dispatch = useDispatch();
+  const { item, options = {} } = props;
+  const [openDialog, setOpenDialog] = React.useState(false);
+  const [visible, setVisible] = React.useState(false);
+  const location = useLocation();
+  const { t } = useTranslation('resource');
+
+  async function updateFunc(numReplicas: number) {
+    try {
+      await item.scale(numReplicas);
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  const applyFunc = React.useCallback(updateFunc, [item]);
+
+  function handleSave(numReplicas: number) {
+    const cancelUrl = location.pathname;
+    const itemName = item.metadata.name;
+
+    setOpenDialog(false);
+
+    // setOpenDialog(false);
+    dispatch(
+      clusterAction(() => applyFunc(numReplicas), {
+        startMessage: t('Scaling {{ itemName }}…', { itemName }),
+        cancelledMessage: t('Cancelled scaling {{ itemName }}.', { itemName }),
+        successMessage: t('Scaled {{ itemName }}.', { itemName }),
+        errorMessage: t('Failed to scale {{ itemName }}.', { itemName }),
+        cancelUrl,
+        errorUrl: cancelUrl,
+        ...options,
+      })
+    );
+  }
+
+  function handleClose() {
+    setOpenDialog(false);
+  }
+
+  React.useEffect(() => {
+    if (item) {
+      item
+        .getAuthorization('update')
+        .then((result: any) => {
+          if (result.status.allowed) {
+            setVisible(true);
+          }
+        })
+        .catch((err: Error) => {
+          console.error(`Error while getting authorization for edit button in ${item}:`, err);
+          setVisible(false);
+        });
+    }
+  }, [item]);
+
+  if (!visible || !['Deployment', 'StatefulSet', 'ReplicaSet'].includes(item.kind)) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      <Tooltip title={t('frequent|Scale') as string}>
+        <IconButton aria-label={t('frequent|scale')} onClick={() => setOpenDialog(true)}>
+          <Icon icon="mdi:overscan" />
+        </IconButton>
+      </Tooltip>
+      <ScaleDialog resource={item} open={openDialog} onClose={handleClose} onSave={handleSave} />
+    </React.Fragment>
+  );
+}
+
+interface ScaleDialogProps extends DialogProps {
+  resource: KubeObject;
+  onSave: (numReplicas: number) => void;
+  onClose: () => void;
+  errorMessage?: string;
+}
+
+const Input = styled(MuiInput)({
+  '& input[type=number]': {
+    MozAppearance: 'textfield',
+    textAlign: 'center',
+  },
+  '& input::-webkit-outer-spin-button, & input::-webkit-inner-spin-button': {
```

Comment:
What is this for, and why webkit only?

Reply:
This hides the up/down arrows from the number type input. It works fine in Firefox too.